### PR TITLE
add pre-start-script job to run arbitrary script

### PR DESCRIPTION
**Pull request by drnic**
Currently using this with `bosh create-env jumpbox-deployment/jumpbox.yml -o prepare_env.yml -v local_os_conf=path/to/os_conf.tgz` where the latter is:

```
- type: replace
  path: /releases/name=os-conf
  value:
    name: os-conf
    version: latest
    url: file://((local_os_conf))

- type: replace
  path: /instance_groups/name=jumpbox/jobs/-
  value:
    name: prepare_env
    release: os-conf
    properties:
      script: |-
          #!/bin/bash

          apt-get update
          apt-get install git ruby wget tmux -y
          wget -q -O - https://raw.githubusercontent.com/starkandwayne/homebrew-cf/master/public.key | apt-key add -
          echo "deb http://apt.starkandwayne.com stable main" | tee /etc/apt/sources.list.d/starkandwayne.list
          apt-get update
          apt-get install bosh-cli credhub-cli cf-cli eden terraform jq direnv -y

          cat > /etc/profile.d/git-aliases.sh <<EOF
          alias gl="git pull"
          alias gp="git push"
          EOF
```

### CODE REVIEW

**Comment by drnic** [comment]
@cppforlife will this PR be ok to merge + cut into a release? Or something you want changed?

File: jobs/prepare_env/templates/pre-start.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -ex
+
```
Comment (cppforlife) [inline jobs/prepare_env/templates/pre-start.sh]: could just use `<%= p('script') %>` in pre-start without making any new files.

File: jobs/prepare_env/spec [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: prepare_env
+
+templates:
+  pre-start.sh: bin/pre-start
+
+properties:
+  script:
+    description: Script that is run during pre-start to allow additional setup of environment, run as root user.
+    example: |-
```
Comment (cppforlife) [inline jobs/prepare_env/spec]: short 2 line example would be nicer imho.

File: jobs/prepare_env/spec [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: prepare_env
```
Comment (cppforlife) [inline jobs/prepare_env/spec]: prepare_env should prolly be called `pre-start-script`

**Comment by drnic** [comment]
@cppforlife thanks for the suggestions; especially the wholesale replacement of `pre-start.sh`. I've confirmed the updates work for my own deployment.

**Comment by dpb587-pivotal** [comment]
[Story 152871915](https://www.pivotaltracker.com/story/show/152871915)

**Pull request merged**